misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java

```diff
@@ -277,9 +277,9 @@ private static List<Node> getRecordFieldsForXMLElement(Element xmlElement, boole
         for (int i = 0; i < xmlAttributesMap.getLength(); i++) {
             org.w3c.dom.Node xmlNode = xmlAttributesMap.item(i);
             if (xmlNode.getNodeType() == org.w3c.dom.Node.ATTRIBUTE_NODE) {
-                if ((xmlNode.getPrefix() == null &&
-                        XMLNS_PREFIX.equals(xmlNode.getLocalName())) || (XMLNS_PREFIX.equals(xmlNode.getPrefix()) &&
-                                xmlNode.getLocalName().equals(xmlElement.getPrefix())) && withNameSpace) {
+                if (((xmlNode.getPrefix() == null && XMLNS_PREFIX.equals(xmlNode.getLocalName())) ||
```

Comment:
This may not relate to this fix, but check why we have this `xmlNode.getPrefix() == null && XMLNS_PREFIX.equals(xmlNode.getLocalName()` check? it seem it always false to me.

Reply:
If we consider the below scenario, the `xmlNode` does not have a prefix and the `localName` of the `xmlNode` is `xmlns`. Hence, this condition will be true.

```
<library xmlns="http://example.com/library">
    <book>
        <title>The Great Gatsby</title>
        <author>F. Scott Fitzgerald</author>
        <genre>Drama</genre>
    </book>
</library>
```